DadMakeAFunny/app/src/main/java/com/example/dadmakeafunny/JokesReader.java

```diff
@@ -0,0 +1,130 @@
+package com.example.dadmakeafunny;
+import android.os.AsyncTask;
+import android.util.Log;
+
+import java.io.InputStream;
+import java.util.ArrayList;
+import android.util.Log;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.ArrayList;
+
+
+public class JokesReader {
+/* class opens connection to reddit and handles all jokes */
+
+    private String url;
+    private ArrayList<Joke> jokes;
+    private String last_joke_id;
+    private static final String BASE_URL = "https://www.reddit.com/r/dadjokes/.json";
+    private static final String LIMIT_FLAG  = "?limit=";
+    private int limit = 20;
+
+    /*constructor*/
+    public JokesReader(){
```

Comment:
Same deal here, i think it would be good to have a constructor that takes in an `ArrayList<Joke>`